## Trabalho de Conclusão de Curso - Análise Exploratória e Modelagem Preditiva do Desempenho em Matemática do 9º ano em escolas públicas com base nos dados do SAEB (2019–2023)
Para a análise exploratória e aplicação de métodos de aprendizado de máquina as seguintes bibliotecas e módulos foram utilizados: 

In [2]:
import os
import pandas as pd
import numpy as np
import sys

### 1. Representação do conjunto de dados
<!-- O dataset primário analisado neste notebook é o Questionário Escolar do Programa de Avaliação Internacional de Estudantes (PISA) 2018 da Organização para Cooperação e Desenvolvimento Econômico (OCDE) (daqui em diante referido como o "Dataset Escolar PISA 2018"), publicado no site da comunidade de ciência de dados Kaggle em 9 de julho de 2021. Ele contém as respostas dos países que responderam ao Questionário Escolar PISA 2018, que abrange áreas relacionadas ao contexto das escolas, sua gestão, seu corpo docente, o desempenho de seus alunos nas avaliações e seus climas, bem como informações sobre grupos específicos, como aprendizes multiculturais (OCDE, 2017). --> 

### 2. Coleta de dados 
Esta seção descreve o processo de coleta de dados realizado para construir os conjuntos de dados, juntamente com as implicações dos métodos de coleta de dados sobre as conclusões e insights gerados ao final da análise de dados.

### 2.1 Estrutura do conjunto de dados
Esta seção descreve as estruturas e conteúdos dos conjuntos de dados utilizados, incluindo o número de linhas e colunas e descrições das observações e variáveis. Eles foram carregados em um pandas.DataFrame, e como já estão fornecidos como um arquivo CSV, podem ser carregados facilmente usando a função read_csv().

#### SAEB 2021

In [6]:
df_saeb_2021 = pd.read_csv("TS_ALUNO_9EF.csv",  encoding='latin-1', sep=";")
df_saeb_2021

C:\Users\emill\AppData\Local\Temp\ipykernel_4032\3911153505.py:1: DtypeWarning: Columns (42,43,44,45,46,47,48,49,50,51) have mixed types. Specify dtype option on import or set low_memory=False.
  df_saeb_2021 = pd.read_csv("TS_ALUNO_9EF.csv",  encoding='latin-1', sep=";")


ID_SAEB  ID_REGIAO  ID_UF  ID_MUNICIPIO  ID_AREA  ID_ESCOLA  \
0           2021          1     11       6316600        2   61324549   
1           2021          1     11       6316600        2   61324549   
2           2021          1     11       6316600        2   61324549   
3           2021          1     11       6316600        2   61324549   
4           2021          1     11       6316600        2   61324549   
...          ...        ...    ...           ...      ...        ...   
2591932     2021          5     53       6322169        1   61397110   
2591933     2021          5     53       6322169        1   61397110   
2591934     2021          5     53       6322169        1   61397110   
2591935     2021          5     53       6322169        1   61397110   
2591936     2021          5     53       6322169        1   61397110   

         IN_PUBLICA  ID_LOCALIZACAO  ID_TURMA  ID_SERIE  ...  TX_RESP_Q21  \
0                 1               1   1324152         9  ...            C   
1                 1               1   1324152         9  ...            B   
2                 1               1   1324152         9  ...            .   
3                 1               1   1324152         9  ...            .   
4                 1               1   1324152         9  ...            C   
...             ...             ...       ...       ...  ...          ...   
2591932           0               1   1475064         9  ...            .   
2591933           0               1   1475064         9  ...            .   
2591934           0               1   1475064         9  ...            .   
2591935           0               1   1475064         9  ...            .   
2591936           0               1   1475064         9  ...            .   

         TX_RESP_Q22a  TX_RESP_Q22b  TX_RESP_Q22c  TX_RESP_Q22d  TX_RESP_Q22e  \
0                   C             D             B             D             B   
1                   B             C             B             C             B   
2                   .             .             .             .             .   
3                   .             .             .             .             .   
4                   D             D             D             D             C   
...               ...           ...           ...           ...           ...   
2591932             .             .             .             .             .   
2591933             .             .             .             .             .   
2591934             .             .             .             .             .   
2591935             .             .             .             .             .   
2591936             .             .             .             .             .   

         TX_RESP_Q22f  TX_RESP_Q22g  TX_RESP_Q22h  TX_RESP_Q22i  
0                   B             B             C             A  
1                   B             B             B             C  
2                   .             .             .             .  
3                   .             .             .             .  
4                   B             D             D             D  
...               ...           ...           ...           ...  
2591932             .             .             .             .  
2591933             .             .             .             .  
2591934             .             .             .             .  
2591935             .             .             .             .  
2591936             .             .             .             .  

[2591937 rows x 144 columns]

Informações gerais sobre o conjunto de dados:
Como ele contém mais de 100 colunas (o valor padrão de display.max_info_columns), o parâmetro verbose de info() foi definido como True para exibir todas as colunas de dados. Pelo mesmo motivo, o parâmetro null_counts também foi explicitamente definido como True.

In [8]:
df_saeb_2021.info(verbose=True, show_counts = True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2591937 entries, 0 to 2591936
Data columns (total 144 columns):
 #    Column                         Non-Null Count    Dtype  
---   ------                         --------------    -----  
 0    ID_SAEB                        2591937 non-null  int64  
 1    ID_REGIAO                      2591937 non-null  int64  
 2    ID_UF                          2591937 non-null  int64  
 3    ID_MUNICIPIO                   2591937 non-null  int64  
 4    ID_AREA                        2591937 non-null  int64  
 5    ID_ESCOLA                      2591937 non-null  int64  
 6    IN_PUBLICA                     2591937 non-null  int64  
 7    ID_LOCALIZACAO                 2591937 non-null  int64  
 8    ID_TURMA                       2591937 non-null  int64  
 9    ID_SERIE                       2591937 non-null  int64  
 10   ID_ALUNO                       2591937 non-null  int64  
 11   IN_SITUACAO_CENSO              2591937 non-null  int64  
 12 

O conjunto de dados contém 2591937 observações (linhas) em 144 variáveis (colunas). Cada observação corresponde a um aluno do 9° ano participante na avaliação em 2021, e as variáveis correspondem a coleta de informações que possam avaliar os eixos do SAEB como equidade (nível socioeconômico dos estudantes, perfil individual e familiar) e atendimento escolar (acesso e trajetória escolar) e desempenho nas provas.
As variáveis e esquemas de codificação para respostas categóricas selecionadas são os seguintes:

In [51]:
dicionario_2021 = pd.read_excel(r"Dicionario_Saeb_2021.xlsx", sheet_name="TS_ALUNO_9EF")
dicionario_2021 = dicionario_2021[["TS_ALUNO_9EF", "Unnamed: 3"]]
dicionario_2021 = dicionario_2021.dropna(how="all")

from IPython.display import display
display(dicionario_2021.style.hide(axis="index"))

Abaixo está uma listagem literal das perguntas do Questionário de Estudante(TX_RESP_Q01 a TX_RESP_Q22i). Os esquemas de codificação para as respostas categóricas também estão enumerados:
#### ID_SAEB
- 2021
#### ID_REGIAO	
- 1: Norte
- 2: Nordeste
- 3: Sudeste
- 4: Sul
- 5: Centro-Oeste
#### ID_UF	
- 11: RO
- 12: AC
- 13: AM
- 14: RR
- 15: PA
- 16: AP
- 17: TO
- 21: MA
- 22: PI
- 23: CE
- 24: RN
- 25: PB
- 26: PE
- 27: AL
- 28: SE
- 29: BA
- 31: MG
- 32: ES
- 33: RJ
- 35: SP
- 41: PR
- 42: SC 
- 43: RS
- 50: MS
- 51: MT
- 52: GO
- 53: DF
#### ID_AREA	
- 1: Capital
- 2: Interior
#### IN_PUBLICA	
- 0: Privada
- 1: Pública
#### ID_LOCALIZACAO	
- 1: Urbana
- 2: Rural
#### ID_SERIE	
- 9: 9° ano do Ensino Fundamental
#### IN_SITUACAO_CENSO
- 0: Não consistente
- 1: Consistente
#### IN_PREENCHIMENTO_LP	
- 0: Prova não prenchida
- 1: Prova preenchida
- #### IN_PREENCHIMENTO_MT
- 0: Prova não prenchida
- 1: Prova preenchida
- #### IN_PREENCHIMENTO_CH	
- 0: Prova não prenchida
- 1: Prova preenchida
- Vazio: Não selecionado para a amostra de CH
- #### IN_PREENCHIMENTO_CN	
- 0: Prova não prenchida
- 1: Prova preenchida
- Vazio: Não selecionado para a amostra de CN
#### IN_PRESENCA_LP	
- 0: Ausente
- 1: Presente
#### IN_PRESENCA_MT
- 0: Ausente
- 1: Presente
#### IN_PRESENCA_CH
- 0: Ausente
- 1: Presente
#### IN_PRESENCA_CN
- 0: Ausente
- 1: Presente
#### ID_CADERNO_LP
- Prova Regular (Cadernos 1 a 21)
- Macrotipo 18 (Caderno 22)
#### ID_BLOCO_1_LP	
- De 1 a 7 (8 no macrotipo 18 - caderno 22)
#### ID_BLOCO_2_LP	
- De 1 a 7 (9 no macrotipo 18 - caderno 22)
#### ID_CADERNO_MT
- Prova Regular (Cadernos 1 a 21)
- Macrotipo 18 (Caderno 22)
#### ID_BLOCO_1_MT	
- De 1 a 7 (8 no macrotipo 18 - caderno 22)
#### ID_BLOCO_2_MT
- De 1 a 7 (9 no macrotipo 18 - caderno 22)
#### ID_CADERNO_CH
- Prova Regular (Cadernos 1 a 26)
- Macrotipo 18 (Caderno 27)
#### ID_BLOCO_1_CH	
- De 1 a 13
#### ID_BLOCO_2_CH	
- De 1 a 13
#### ID_BLOCO_3_CH	
- De 1 a 13
#### NU_BLOCO_1_ABERTA_CH	
- De 1 a 6
#### NU_BLOCO_2_ABERTA_CH
- De 1 a 6
#### ID_CADERNO_CN	
- Prova Regular (Cadernos 1 a 14)
- Macrotipo 18 (Caderno 15)
#### ID_BLOCO_1_CN	
- De 1 a 7
#### ID_BLOCO_2_CN	
- De 1 a 7
#### ID_BLOCO_3_CN	
- De 1 a 7
#### NU_BLOCO_1_ABERTA_CN	
- De 1 a 3
#### NU_BLOCO_2_ABERTA_CN
- De 1 a 3
#### TX_RESP_BLOCO_1_LP	
- A, B, C, D, . (branco), * (nulo)
#### TX_RESP_BLOCO_2_LP	
- A, B, C, D, . (branco), * (nulo)
#### TX_RESP_BLOCO_1_MT	
- A, B, C, D, . (branco), * (nulo)
#### TX_RESP_BLOCO_2_MT	
- A, B, C, D, . (branco), * (nulo)
#### TX_RESP_BLOCO_1_CH
- A, B, C, D, . (branco), * (nulo)
#### TX_RESP_BLOCO_2_CH	
- A, B, C, D, . (branco), * (nulo)
#### TX_RESP_BLOCO_3_CH	
- A, B, C, D, . (branco), * (nulo)
#### CO_CONCEITO_Q1_CH	
- 0: Nenhum crédito
- 1: Crédito parcial
- 2: Crédito total
- 7: Erros de impressão ou digitalização
- .: Branco
#### CO_CONCEITO_Q2_CH	
- 0: Nenhum crédito
- 1: Crédito parcial
- 2: Crédito total
- 7: Erros de impressão ou digitalização
- .: Branco
#### TX_RESP_BLOCO1_CN	
- A, B, C, D, . (branco), * (nulo)
#### TX_RESP_BLOCO2_CN	
- A, B, C, D, . (branco), * (nulo)
#### TX_RESP_BLOCO3_CN	
- A, B, C, D, . (branco), * (nulo)
#### CO_CONCEITO_Q1_CN	
- 0: Nenhum crédito
- 1: Crédito parcial
- 2: Crédito total
- 7: Erros de impressão ou digitalização
- .: Branco
#### CO_CONCEITO_Q2_CN	
- 0: Nenhum crédito
- 1: Crédito parcial
- 2: Crédito total
- 7: Erros de impressão ou digitalização
- .: Branco
#### IN_PROFICIENCIA_LP	
- 0: Não
- 1: Sim
#### IN_PROFICIENCIA_MT	
- 0: Não
- 1: Sim
#### IN_PROFICIENCIA_CH
- 0: Não
- 1: Sim
#### IN_PROFICIENCIA_CN	
- 0: Não
- 1: Sim
#### IN_AMOSTRA
- 0: Não
- 1: Sim
#### ESTRATO	
- Os estratos são compostos por características da participação da escola na avaliação e representam agrupamentos para os quais a avaliação fornece resultados confiáveis para Língua Portuguesa e Matemática. 
#### ESTRATO_CIENCIAS
- Os estratos são compostos por características da participação da escola na avaliação e representam agrupamentos para os quais a avaliação fornece resultados confiáveis para Ciências Humanas e Ciências da Natureza. 
#### PESO_ALUNO_LP
- Valor com 7 casas decimais 
#### PROFICIENCIA_LP	
- Valor com 7 casas decimais 
#### ERRO_PADRAO_LP
- Valor com 7 casas decimais 
#### PROFICIENCIA_LP_SAEB	
- Valor com 7 casas decimais 
#### ERRO_PADRAO_LP_SAEB	
- Valor com 7 casas decimais 
#### PESO_ALUNO_MT
- Valor com 7 casas decimais 
#### PROFICIENCIA_MT	
- Valor com 7 casas decimais 
#### ERRO_PADRAO_MT	
- Valor com 7 casas decimais 
#### PROFICIENCIA_MT_SAEB	
- Valor com 7 casas decimais 
#### ERRO_PADRAO_MT_SAEB
- Valor com 7 casas decimais 
#### PESO_ALUNO_CH	
- Valor com 7 casas decimais 
#### PROFICIENCIA_CH
- Valor com 7 casas decimais 
#### ERRO_PADRAO_CH	
- Valor com 7 casas decimais 
#### PROFICIENCIA_CH_SAEB	
- Valor com 7 casas decimais 
#### ERRO_PADRAO_CH_SAEB	
- Valor com 7 casas decimais 
#### PESO_ALUNO_CN
- Valor com 7 casas decimais 
#### PROFICIENCIA_CN
- Valor com 7 casas decimais 
#### ERRO_PADRAO_CN
- Valor com 7 casas decimais 
#### PROFICIENCIA_CN_SAEB
- Valor com 7 casas decimais 
#### ERRO_PADRAO_CN_SAEB
- Valor com 7 casas decimais
#### IN_PREENCHIMENTO_QUESTIONARIO
- 0: Não preenchido
- 1: Preenchido parcial ou totalmente
#### IN_INSE	
- 0: Não
- 1: Sim
#### NU_TIPO_NIVEL_INSE	
- 1: Nível I
- 2: Nível II
- 3: Nível III
- 4: Nível IV
- 5: Nível V
- 6: Nível VI
- 7: Nível VII
- 8: Nível VIII
#### PESO_ALUNO_INSE
- Valor com 7 casas decimais
- #### TX_RESP_Q01	- Qual é o seu sexo?
- *: Nulo
- .: Branco
- A: Masculino
- B: Feminino
#### TX_RESP_Q02 - Qual é a sua idade?
- *: Nulo
- .: Branco
- A: 13 anos ou menos
- B: 14 anos
- C: 15 anos
- D: 16 anos
- E: 17 anos
- F: 18 anos ou mais
#### TX_RESP_Q03 - Qual língua seus pais falam com mais frequência em casa?
- *: Nulo
- .: Branco
- A: Português
- B: Espanhol
- C: Outra língua
#### TX_RESP_Q04 - Qual é a sua cor ou raça?
- *: Nulo
- .: Branco
- A: Branca
- B: Preta
- C: Parda
- D: Amarela
- E: Indígena
- F: Não quero declarar
#### TX_RESP_Q05 - Você possui algum tipo de necessidade especial?
- *: Nulo
- .: Branco
- A: Sim
- B: Não
#### TX_RESP_Q06a - Normalmente, quem mora na sua casa? - Mãe ou madrasta
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q06b - Normalmente, quem mora na sua casa? - Pai ou padrasto
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q06c -  Normalmente, quem mora na sua casa? - Irmão(s) ou irmã(s)
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q06d - Normalmente, quem mora na sua casa? - Avô ou avó
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q06e -  Normalmente, quem mora na sua casa? - Outros (tios, primos, etc.)
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q07 - Qual é a maior escolaridade da sua mãe (ou mulher responsável por você)?
- *: Nulo
- .: Branco
- A: Não completou o 5° ano do Ensino Fundamental
- B: Ensino Fundamental, até o 5° ano
- C: Ensino Fundamental completo
- D: Ensino Médio completo
- E: Ensino Superior completo (faculdade ou graduação)
- F: Não sei
#### TX_RESP_Q08 - Qual é a maior escolaridade do seu pai (ou homem responsável por você)?
- *: Nulo
- .: Branco
- A: Não completou o 5° ano do Ensino Fundamental
- B: Ensino Fundamental, até o 5° ano
- C: Ensino Fundamental completo
- D: Ensino Médio completo
- E: Ensino Superior completo (faculdade ou graduação)
- F: Não sei
#### TX_RESP_Q09a - Com que frequência seus pais ou responsáveis costumam: - Ler em casa
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B De vez em quando
- C: Sempre ou quase sempre
#### TX_RESP_Q09b -  Com que frequência seus pais ou responsáveis costumam: - Conversar com você sobre o que acontece na escola
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B De vez em quando
- C: Sempre ou quase sempre
#### TX_RESP_Q09c -  Com que frequência seus pais ou responsáveis costumam: - Incentivar você a estudar
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B De vez em quando
- C: Sempre ou quase sempre
#### TX_RESP_Q09d -  Com que frequência seus pais ou responsáveis costumam: - Incentivar você a fazer a tarefa de casa
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B De vez em quando
- C: Sempre ou quase sempre
#### TX_RESP_Q09e -  Com que frequência seus pais ou responsáveis costumam: - Incentivar você a comparecer às aulas
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B De vez em quando
- C: Sempre ou quase sempre
#### TX_RESP_Q09f -  Com que frequência seus pais ou responsáveis costumam: - Ir às reuniões de pais na escola
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B De vez em quando
- C: Sempre ou quase sempre
#### TX_RESP_Q10a - Na rua que você mora tem: Asfalto ou calçamento
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q10b - Na rua que você mora tem: Água tratada
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q10c - Na rua que você mora tem: Iluminação
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q11a - Dos itens relacionados abaixo, quantos existem na sua casa? - Geladeira
- *: Nulo
- .: Branco
- A: Nenhum
- B: 1
- C: 2
- D: 3 ou mais
#### TX_RESP_Q11b - Dos itens relacionados abaixo, quantos existem na sua casa? - Tablet
- *: Nulo
- .: Branco
- A: Nenhum
- B: 1
- C: 2
- D: 3 ou mais
#### TX_RESP_Q11c - Dos itens relacionados abaixo, quantos existem na sua casa? - Computador (ou notebook)
- *: Nulo
- .: Branco
- A: Nenhum
- B: 1
- C: 2
- D: 3 ou mais
#### TX_RESP_Q11d - Dos itens relacionados abaixo, quantos existem na sua casa? - Quartos para dormir
- *: Nulo
- .: Branco
- A: Nenhum
- B: 1
- C: 2
- D: 3 ou mais
#### TX_RESP_Q11e - Dos itens relacionados abaixo, quantos existem na sua casa? - Televisão
- *: Nulo
- .: Branco
- A: Nenhum
- B: 1
- C: 2
- D: 3 ou mais
#### TX_RESP_Q11f - Dos itens relacionados abaixo, quantos existem na sua casa? - Banheiro
- *: Nulo
- .: Branco
- A: Nenhum
- B: 1
- C: 2
- D: 3 ou mais
#### TX_RESP_Q11g - Dos itens relacionados abaixo, quantos existem na sua casa? - Carro
- *: Nulo
- .: Branco
- A: Nenhum
- B: 1
- C: 2
- D: 3 ou mais
#### TX_RESP_Q11h - Dos itens relacionados abaixo, quantos existem na sua casa? - Celular com internet (smartphone)
- *: Nulo
- .: Branco
- A: Nenhum
- B: 1
- C: 2
- D: 3 ou mais
#### TX_RESP_Q12a - Na sua casa tem: - Tv por internet (Netflix, GloboPlay, etc.)
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q12b - Na sua casa tem: - Rede Wi-Fi
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q12c - Na sua casa tem: - Um quarto só seu
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q12d - Na sua casa tem: - Mesa para estudar
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q12e - Na sua casa tem: - Forno de microondas
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q12f - Na sua casa tem: - Aspirador de pó
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q12g - Na sua casa tem: - Máquina de lavar roupa 
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q12h - Na sua casa tem: - Freezer (independente ou segunda porta da geladeira)
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q12i - Na sua casa tem: - Garagem
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q13 - Quanto tempo você demora para chegar à sua escola?
- *: Nulo
- .: Branco
- A: Menos de 30 minutos
- B: Entre 30 minutos e uma hora
- C: Mais de uma hora
#### TX_RESP_Q14 - Considerando a maior distãncia percorrida, normalmente de que forma você chega à sua escola?
- *: Nulo
- .: Branco
- A: À pé
- B: De bicicleta 
- C: De Van (ou Kombi)
- D: De ônibus
- E: De metrô (ou trem urbano)
- F: De carro
- G: Outros meios de transporte (barco, motocicleta, etc.)
#### TX_RESP_Q15 - Você se utiliza de transporte escolar, ou passe escolar, para ir à escola?
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q16 - Com que idade você entrou na escola?
- *: Nulo
- .: Branco
- A: 3 anos ou menos
- B: 4 ou 5 anos 
- C: 6 ou 7 anos
- D: 8 anos ou mais
#### TX_RESP_Q17 - A partir do primeiro ano do ensino fundamental, em que tipo de escola você estudou?
- *: Nulo
- .: Branco
- A: Somente em escola pública
- B: Somente em escola particular
- C: Em escola pública e em escola particular
#### TX_RESP_Q18 - Você já foi reprovado(a)?
- *: Nulo
- .: Branco
- A: Não
- B: Sim, uma vez
- C: Sim, duas vezes ou mais
#### TX_RESP_Q19 - Alguma vez você abandonou a escola deixando de frequentá-la até o final do ano escolar?
- *: Nulo
- .: Branco
- A: Nunca
- B: Sim, uma vez
- C: Sim, duas vezes ou mais
#### TX_RESP_Q20a - Fora da escola em dias de aula, quanto tempo você usa para: - Estudar (lição de casa, trabalhos escolares, etc.)
- *: Nulo
- .: Branco
- A: Não uso meu tempo pra isso
- B: Menos de 1 hora
- C: Entre 1 e 2 horas
- D: Mais de 2 horas
#### TX_RESP_Q20b - Fora da escola em dias de aula, quanto tempo você usa para: - Fazer cursos
- *: Nulo
- .: Branco
- A: Não uso meu tempo pra isso
- B: Menos de 1 hora
- C: Entre 1 e 2 horas
- D: Mais de 2 horas
#### TX_RESP_Q20c - Fora da escola em dias de aula, quanto tempo você usa para: - Trabalhar em casa (lavar louça, limpar quintal, cuidar dos irmãos)
- *: Nulo
- .: Branco
- A: Não uso meu tempo pra isso
- B: Menos de 1 hora
- C: Entre 1 e 2 horas
- D: Mais de 2 horas
#### TX_RESP_Q20d - Fora da escola em dias de aula, quanto tempo você usa para: - Trabalhar fora de casa (recebendo ou não um salário)
- *: Nulo
- .: Branco
- A: Não uso meu tempo pra isso
- B: Menos de 1 hora
- C: Entre 1 e 2 horas
- D: Mais de 2 horas
#### TX_RESP_Q20e - Fora da escola em dias de aula, quanto tempo você usa para: - Lazer (TV, internet, brincar, música, etc.)
- *: Nulo
- .: Branco
- A: Não uso meu tempo pra isso
- B: Menos de 1 hora
- C: Entre 1 e 2 horas
- D: Mais de 2 horas
#### TX_RESP_Q21 - Quando terminar o Ensino Fundamental você pretende
- *: Nulo
- .: Branco
- A: Somente continuar estudando
- B: Somente trabalhar
- C: Continuar estudando e trabalhar
- D: Ainda não sei
#### TX_RESP_Q22a - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Eu possuía equipamento adequado para acompanhar o ensino remoto
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre
#### TX_RESP_Q22b - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Eu tive conexão de internet adequada para acesso às aulas remotas
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre
#### TX_RESP_Q22c - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Tive facilidade em usar os programas de comunicação nas aulas remotas
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre
#### TX_RESP_Q22d - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Recebi material impresso da escola (livros, apostilas, atividades em folha, etc.) 
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre
#### TX_RESP_Q22e - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Os professores me auxiliaram a entender o conteúdo
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre
#### TX_RESP_Q22f - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Eu compreendi o conteúdo das aulas remotas
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre
#### TX_RESP_Q22g - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Em casa havia um lugar tranquilo para eu assistir às aulas
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre
#### TX_RESP_Q22h - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Meus familiares apoiaram o meu estudo
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre
#### TX_RESP_Q22i - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Meus colegas me apoiaram durante o ensino remoto
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre

### 3. Pré-processamento dos dados
Esta seção descreve o pré-processamento realizado nos dados antes de submetê-los à análise. Em particular, as motivações e os processos relacionados à fusão de conjuntos de dados e à limpeza de dados são discutidos.